app/runtime/runtime/src/main/java/io/syndesis/integration/runtime/LoggingInterceptStrategy.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.camel.spi.InterceptStrategy;
+import org.apache.camel.spi.LogListener;
+import org.apache.camel.spi.UuidGenerator;
+import org.apache.camel.util.CamelLogger;
+
+import java.io.StringWriter;
+
+/**
+ * Created by chirino on 1/10/18.
+ */
+public class LoggingInterceptStrategy implements InterceptStrategy {
+
+
+    public UuidGenerator getUuidGenerator() {
+        return new UuidGenerator() {
+            @Override
+            public String generateUuid() {
+                // Generates always incrementing lexically sortable unique keys
+                return KeyGenerator.createKey();
+            }
+        };
+    }
+
+    public static class LogLine {
+        public final String id;
+        public final String tx;
+        public final String step;
+        public final String action;
+        public final String message;
+
+        public LogLine(String tx, String step, String action, String message) {
+            this.id = KeyGenerator.createKey();
+            this.tx = tx;
+            this.step = step;
+            this.action = action;
+            this.message = message;
+        }
+
+        @Override
+        public String toString() {
+
+            try {
+
+                StringWriter w = new StringWriter();
+                JsonGenerator jg = new JsonFactory().createGenerator(w);
+                jg.writeStartObject();
+                if( id!=null ) {
+                    jg.writeFieldName("id");
+                    jg.writeString(id);
+                }
+                if( tx!=null ) {
+                    jg.writeFieldName("tx");
+                    jg.writeString(tx);
+                }
+                if( step!=null ) {
+                    jg.writeFieldName("step");
+                    jg.writeString(step);
+                }
+                if( action!=null ) {
+                    jg.writeFieldName("action");
+                    jg.writeString(action);
+                }
+                if( message!=null ) {
+                    jg.writeFieldName("message");
+                    jg.writeString(message);
+                }
+
+                jg.writeEndObject();
+                jg.close();
+                return w.toString();
+            } catch (java.io.IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+
+    @Override
+    public Processor wrapProcessorInInterceptors(CamelContext context, ProcessorDefinition<?> definition, Processor target, Processor nextTarget) throws Exception {
+        // don't trace begin/end for logging steps.
+        if( definition instanceof LogDefinition ) {
+            return target;
+        }
+        return exchange -> {
+
+            String tx = exchange.getExchangeId();
+            String step = definition.getId();
+
+            System.out.println(new LogLine(tx, step, "begin", null));
+            target.process(exchange);
+            System.out.println(new LogLine(tx, step, "end", null));
```

Comment:
Wondering if we could avoid creating this LogLine object and instead have a static utility method that prints directly to thread local writer (or somethign similar)

Reply:
Is this for performance reasons?

Reply:
Just to avoid useless temporary object to be allocated and destroyed

Reply:
Lets optimize once we get a performance testing framework.